test/functional/page_objects/discover_page.js

```diff
@@ -74,12 +66,8 @@ export function DiscoverPageProvider({ getService, getPageObjects }) {
 
     async openSavedSearch() {
       await this.clickLoadSavedSearchButton();
-      await retry.try(async () => {
-        const isLoadFormVisible = await testSubjects.exists('loadSearchForm');
-        if (!isLoadFormVisible) {
-          throw new Error('Load search form not visible yet.');
-        }
-      });
+      await testSubjects.exists('loadSearchForm');
+      await PageObjects.header.waitUntilLoadingHasFinished();
```

Comment:
I am rather sure, that won't be enough to fix the issue with the flakyness. Unfortunately we don't have any failed test runs anymore, but in the ones I debugged, that had that failure, the `loadSearchForm` wasn't even present in the HTML, i.e. the button has been clicked, but the panel never properly expanded. So just waiting for something after the panel is there, won't help against that flakyness.

Reply:
Looking at the huge amount of test passes in this PR, maybe it fixed it after all.. :thinking: